docs/readme.md

```diff
@@ -0,0 +1,212 @@
+# Notes:
+```
+This read me file is going to contain everything I learned while on my placement with RJJ Software
+```
+## Git and gitflow:
+```
+Git is a source control system. It allows us to track all changes to a given code automatically.
+```
+[Atlassian - Learn Git](https://www.atlassian.com/git#:~:text=Here%20is%20a%20basic%20overview,commit%22%20(save)%20the%20changes)
+
+Basic overview of how Git works:
+```
+1. Create a repository (project) with a git hosting tool (GitHub)
+2. Copy/clone the repository to your local machine
+3. Add a file to your local repo and "commit" save the changes
+4. Push the changes to your main branch
+5. Make a change to your file with a git hosting tool and commit
+6. "Pull" the changes to your local machine
+7. Create a "branch" (version), make a change, commit the change
+8. Open a "pull" request (propose the changes to the main branch)
+9. Merge your branch to the main branch
+```
+List above explained further:
+```
+The basic idea is that when we create a git repo, there is a main branch. This represents the current state of play for the code within the code base. When we want to make a change to the code base, we create a branch - this is a copy of the code up to that point - and work on it until we have completed the work.
+Changes amde to the code base are committed into the branch, this ensures that related changes are saved into source control at the same time. Each commit requires a commmit message, and this should be a helpful description of what the changes are and how they effect the code base.

```

Comment:
typo here (amde -> made). It's a nitpick, but in industry you'll be forced to check you spelling on things.

If you use Visual Studio Code to check your markdown files, you can install the [Docs Authoring Pack](https://marketplace.visualstudio.com/items?itemName=docsmsft.docs-authoring-pack) extension, and it will highlight any typos for you.